scripts/utils/price_utils.js

```diff
@@ -118,19 +125,78 @@ const getDexagPrice = async function (tokenBought, tokenSold, globalPriceStorage
   return price
 }
 
+/**
+ * Returns the price on the 1inh aggregator for a given pair.
+ * Optionally, it first checks whether the price is present in an input cache and
+ * in case returns that price without any network request.
+ *
+ * Note that the output price is not very accurate, especially for tokens with
+ * very high value and very little liquidity. This is due to the fact that 1inch
+ * returns a sell and a by price for a given amount, unlike dex.ag, and this function returns only
+ * one of the two.
+ *
+ * See https://1inch.exchange/#/api for API documentation.
+ *
+ * @param {TokenObject} baseToken base token for the price
+ * @param {TokenObject} quoteToken quote token for the price
+ * @param {object} [globalPriceStorage=null] object linking token pairs to prices
+ * @returns {number|undefined} desired price if available, undefined otherwise
+ */
+const getOneinchPrice = async function (baseToken, quoteToken, globalPriceStorage = null) {
+  // TODO: escape `-` in token symbols
+  if (globalPriceStorage !== null && baseToken + "-" + quoteToken in globalPriceStorage) {
+    return globalPriceStorage[baseToken + "-" + quoteToken]
+  }
+  if (globalPriceStorage !== null && quoteToken + "-" + baseToken in globalPriceStorage) {
+    return 1.0 / globalPriceStorage[quoteToken + "-" + baseToken]
+  }
+  // use ETH instead of WETH for a more reliable price
+  baseToken = baseToken == "WETH" ? "ETH" : baseToken
+  quoteToken = quoteToken == "WETH" ? "ETH" : quoteToken
+  // 1inch returns only buy and sell prices for specific
+  const quoteTokenAmount = toErc20Units(TEN.pow(new BN(quoteToken.decimals)), quoteToken.decimals)
```

Comment:
Should it be: toErc20Units(TEN, quoteToken.decimals)? I am confused right now.

Should we maybe use 1, instead of 10, just because selling 10BTC could already influence the price quite a bit?

Reply:
Whoops! That's absolutely a mistake, very good you spotted it!